usecase/utils.go

```diff
@@ -0,0 +1,140 @@
+package usecase
+
+import (
+	"api/entities"
+	"encoding/csv"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math"
+	"os"
+	"strconv"
+	"sync/atomic"
+)
+
+var readerCounter int32
+
+func ReadCsv(filename string) ([][]string, error) {
+	file, err := os.Open(filename)
+	if err != nil {
+		return nil, err
+	}
+	reader := csv.NewReader(file)
+	records, err := reader.ReadAll()
+	if err != nil {
+		return nil, err
+	}
+	return records, nil
+}
+
+func ParseRecordsToJSON(records [][]string) ([]byte, error) {
+	recordsMap, err := ParseRecordsToMap(records)
+	if err != nil {
+		return nil, err
+	}
+	response, err := json.Marshal(recordsMap)
+	if err != nil {
+		return nil, err
+	}
+	return response, nil
+}
+
+func ParseRecordsToMap(records [][]string) (map[int]string, error) {
+	recordsMap := make(map[int]string)
+	for _, line := range records {
+		id, err := strconv.Atoi(line[0])
+		if err != nil {
+			return nil, err
+		}
+		recordsMap[id] = line[1]
+	}
+	return recordsMap, nil
+}
+
+func WriteCSVFromResponse(filename string, data *entities.Response) error {
+	csvFile, err := os.Create(filename)
+	if err != nil {
+		return err
+	}
+	defer csvFile.Close()
+
+	writer := csv.NewWriter(csvFile)
+	for index, berry := range data.Results {
+		var row []string
+		row = append(row, strconv.Itoa(index))
+		row = append(row, berry.Name)
+		row = append(row, berry.Url)
+		writer.Write(row)
+	}
+	writer.Flush()
+	return nil
+}
+
+func ValidateConcurrentReadParams(readType string, items int, itemsPerWorker int) (int32, error) {
+	var initialCounterPosition int32
+	var err error
+	if readType == "odd" {
+		initialCounterPosition = 1
+	} else if readType == "even" {
+		initialCounterPosition = 0
+	} else {
+		err = errors.New("Invalid type")
```

Comment:
Since there's no API documentation describing the params that you're expecting to receive, you can be a bit more explicit here, specifying the types that you're expecting.